src/main/java/org/inferred/freebuilder/processor/Metadata.java

```diff
@@ -135,6 +135,12 @@ public ParameterizedType getBuilder() {
   /** Returns metadata about the properties of the type. */
   public abstract ImmutableList<Property> getProperties();
 
+  /** Returns super types implementing builder */
+  public abstract ImmutableSet<ParameterizedType> getSuperBuilderTypes();
+
+  /** Returns super type properties */
+  public abstract ImmutableMap<ParameterizedType, ImmutableList<Property>> getSuperTypeProperties();
```

Comment:
You'll need to factor in visibility as well; we can't accidentally generate a method accepting a package-scoped type from another package, as that's a compile error.